specification/dartLangSpec.tex

```diff
@@ -14866,19 +14947,12 @@ \subsubsection{Asynchronous For-in}
 the subscription $u$ is canceled by evaluating \code{\AWAIT{} v.cancel()}
 where $v$ is a fresh variable referencing the stream subscription $u$.
 If that evaluation throws,
-execution of $f$ throws the same exception object and stack trace.
-Otherwise execution of $f$ throws with $e$ as exception object and $st$ as stack trace.
-
-\LMHash{}%
-When $u$ is done, execution of $f$ completes normally.
+execution of $s_b$ throws the same exception object and stack trace.
+Otherwise execution of $s_b$ throws
+with $e$ as exception object and $st$ as stack trace.
 
 \LMHash{}%
-It is a compile-time error if an asynchronous for-in statement appears inside a synchronous function (\ref{functions}).
-It is a compile-time error if a traditional for loop (\ref{forLoop}) is prefixed by the \AWAIT{} keyword.
-
-\rationale{
-An asynchronous loop would make no sense within a synchronous function, for the same reasons that an await expression makes no sense in a synchronous function.
-}
+When $u$ is done, execution of $s_b$ and $s$ completes normally.
```

Comment:
Just *s*. During execution of *s_b*, there is no way *u* can be done because it just *synchronously* sent a data event, and either *s_b* completes synchronously, or it pauses *u* which prevents it from being done (can't send a "done" event).

Reply:
OK, done!